public/app/core/services/context_srv.ts

```diff
@@ -25,6 +27,16 @@ export class User {
   }
 }
 
+/**
+ * UserPermission is a map storing permissions in a form of
+ * {
+ *   action: { scope: scope }
+ * }
+ */
+export type UserPermission = {
```

Comment:
Should this be in the "app/types" instead?

Reply:
Yes, makes sense to move.